java/bom/pom.xml

```diff
@@ -0,0 +1,160 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Licensed to the Apache Software Foundation (ASF) under one or more contributor
+  license agreements. See the NOTICE file distributed with this work for additional
+  information regarding copyright ownership. The ASF licenses this file to
+  You under the Apache License, Version 2.0 (the "License"); you may not use
+  this file except in compliance with the License. You may obtain a copy of
+  the License at http://www.apache.org/licenses/LICENSE-2.0 Unless required
+  by applicable law or agreed to in writing, software distributed under the
+  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+  OF ANY KIND, either express or implied. See the License for the specific
+  language governing permissions and limitations under the License. -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.apache</groupId>
+    <artifactId>apache</artifactId>
+    <version>18</version>
```

Comment:
I guess because that's what our root POM uses still: https://github.com/apache/arrow/blob/67cef6416fcf5654aef29394e969d7510e49f4b2/java/pom.xml#L15-L19

I don't think we have any reason to stick to it if there's a much newer version, but if there are potential major changes we could open a separate PR to bump things afterwards?

Reply:
Fair point :)
I propose to keep apache 18 in this PR.
I will do another PR to upgrade to apache 30 (coupling with other updates as said in my previous comment, I should be able to submit several PRs tonight or tomorrow my time 😄  ).

Thanks !